src/proposals/proposal.cpp

```diff
@@ -125,61 +129,32 @@ namespace hypha
         }
     }
 
-    name Proposal::registerBallot(const name &proposer,
+    eosio::checksum256 Proposal::registerBallot(const name &proposer,
                                   const string &title, const string &description, const string &content)
     {
         check(has_auth(proposer) || has_auth(m_dao.get_self()), "Authentication failed. Must have authority from proposer: " +
                                                                     proposer.to_string() + "@active or " + m_dao.get_self().to_string() + "@active.");
 
-        // increment the ballot_id
-        name ballotId = name(m_dao.getSettingOrFail<eosio::name>(LAST_BALLOT_ID).value + 1);
-        m_dao.setSetting(LAST_BALLOT_ID, ballotId);
-
-        name trailContract = m_dao.getSettingOrFail<eosio::name>(TELOS_DECIDE_CONTRACT);
-
-        trailservice::trail::ballots_table b_t(trailContract, trailContract.value);
-        auto b_itr = b_t.find(ballotId.value);
-        check(b_itr == b_t.end(), "ballot_id: " + ballotId.to_string() + " has already been used.");
-
-        vector<name> options;
-        options.push_back(name("pass"));
-        options.push_back(name("fail"));
-        options.push_back(name("abstain"));
-
-        action(
-            permission_level{m_dao.get_self(), name("active")},
-            trailContract, name("newballot"),
-            std::make_tuple(
-                ballotId,
-                name("poll"),
-                m_dao.get_self(),
-                common::S_HVOICE,
-                name("1token1vote"),
-                options))
-            .send();
-
-        action(
-            permission_level{m_dao.get_self(), name("active")},
-            trailContract, name("editdetails"),
-            std::make_tuple(
-                ballotId,
-                title,
-                description.substr(0, std::min(description.length(), size_t(25))),
-                content))
-            .send();
-
         auto expiration = time_point_sec(current_time_point()) + m_dao.getSettingOrFail<int64_t>(VOTING_DURATION_SEC);
 
-        action(
-            permission_level{m_dao.get_self(), name("active")},
-            trailContract, name("openvoting"),
-            std::make_tuple(ballotId, expiration))
-            .send();
-
-        return ballotId;
+        ContentGroups contentGroups{
+            ContentGroup{
+                Content(CONTENT_GROUP_LABEL, "settings"),
+                Content("expiration", expiration)
+            },
+            ContentGroup{
+                Content(CONTENT_GROUP_LABEL, "options"),
+                Content("name", name("pass")),
+                Content("fail", name("fail")),
+                Content("abstain", name("abstain")),
+            }
+        };
```

Comment:
Do you have any suggestion for this structure? I think the "usage" is correct, but haven't really tested it.